app/models/user_step.rb

```diff
@@ -18,6 +18,18 @@ def refresh_status
     end
   end
 
+  def update_status(new_status:)
+    update(status: new_status, submitted_at: new_status == 'waiting' ? Date.today : nil)
```

Comment:
mozno lepsie tu dat `waiting_until` ako nejaky datum kedy to vyprsi (nebude potom treba tolko pocitat). 

lebo pri nejakom podani ti mozu lehotu aj posunut - cize je to od pripadu k pripadu.

Reply:
Zatial som to nechal tak, aby tam ostala informacia kedy to bolo podane, mozno sa na nieco hodi.